src/components/TodoList/TodosList.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useEffect, useMemo } from 'react';
+import classNames from 'classnames';
+import './TodoList.scss';
+import { useDispatch, useSelector } from 'react-redux';
+import { getTodos } from '../../api/todos';
+import {
+  loadTodosActions,
+  loadUserAction,
+  setInputValue,
+  setSelectValue,
+} from '../../store/actions';
+import {
+  getTodosSelector,
+  getUserSelector,
+  getInputValue,
+  getSelectValue,
+} from '../../store/selectors';
+import { getUser } from '../../api/user';
+
+export const TodosList: React.FC = () => {
+  const dispatch = useDispatch();
+
+  const todos = useSelector(getTodosSelector);
+  const user = useSelector(getUserSelector);
+  const inputValue = useSelector(getInputValue);
+  const selectValue = useSelector(getSelectValue);
+
+  const handelSelectUser = async (userId: number) => {
+    if (userId !== user?.id) {
+      const userFromServer = await getUser(userId);
+
+      dispatch(loadUserAction(userFromServer));
+    }
+  };
+
+  const filterTodos = (): Todo[] => {
+    let filteredTodos;
+
+    switch (selectValue) {
+      case 'notCompleted':
+        filteredTodos = todos.filter(todo => !todo.completed);
+        break;
+
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed);
+        break;
+
+      default:
+        filteredTodos = [...todos];
+    }
+
+    return filteredTodos.filter(todo => (
+      todo.title.toLowerCase().includes(inputValue.toLocaleLowerCase())
+    ));
+  };
+
+  const filteredTodos = useMemo(filterTodos, [inputValue, selectValue, todos]);
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    dispatch(setSelectValue(event.target.value));
+  };
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    dispatch(setInputValue(event.target.value));
+    filterTodos();
```

Comment:
redundant, your filtered todos variable will recalculate it's value on inputChange anyways
```suggestion
```